trino/client.py

```diff
@@ -288,11 +289,13 @@ def __init__(
         request_timeout: Union[float, Tuple[float, float]] = constants.DEFAULT_REQUEST_TIMEOUT,
         handle_retry=exceptions.RetryWithExponentialBackoff(),
         verify: bool = True,
+        role: Optional[str] = None,
     ) -> None:
         self._client_session = client_session
         self._host = host
         self._port = port
         self._next_uri: Optional[str] = None
+        self._role = role
```

Comment:
role should be applied onto `ClientSession`, the set role will apply to the connection and must be thread-safe.

`ClientSession` is initialized in the `dbapi.Connection` and further modified based on the headers coming back from the server.

See how schema and catalog are modified as an example.

Reply:
Sure thing. Thanks for the example